tests/publish.bats

```diff
@@ -93,6 +93,32 @@ setup() {
     assert_output --partial "Cache url is required!"
 }
 
+# Publish requires a cached binary.
+# If the binary is not found at the first try, publish will retry for 3 times
+@test "flox publish retries fetching url 3 times" {
+    run $FLOX_CLI -v publish "$CHANNEL#hello" --public-cache-url http://url.example
+    assert_failure
+    assert_output --regexp - <<EOF
+.*
+Checking binary can be downloaded from http://url\.example/\.\.\.
+.*
+(Unable to find binary at http://url\.example/: Failed to invoke path-info:.*){4}
+EOF
+}
+
+# Publish requires a cached binary.
+# If the binary is not found at the first try, publish will retry for `--retry <n>` times
+@test "flox publish retries fetching url; --retry 2" {
+    run $FLOX_CLI -v publish "$CHANNEL#hello" --public-cache-url http://url.example --retry 2
+    assert_failure
+    assert_output --regexp - <<EOF
+.*
+Checking binary can be downloaded from http://url\.example/\.\.\.
+.*
+(Unable to find binary at http://url\.example/: Failed to invoke path-info:.*){3}
```

Comment:
Is `Failed to invoke path-info: .*` something we should be displaying to the user?

Reply:
I just display the error currently, I guess we do want to show the nix output somewhere.
Would it be preferable logging those as debug?